Update Skeleton Instructions for Development Guide Project Setup (#97)
cisagov/development-guide: project_setup/skeletonize-existing-repository.md

# Skeletonize an Existing Repository #

> Skeletonizing a repository standardizes our development setup and environment,

Reviewer: ```suggestion
Skeletonizing a repository standardizes our development setup and environment
```

Author: Hi @mcdonnnj , removed comma at end. of line. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

and enables our [Lineage GitHub Action](https://github.com/cisagov/action-lineage/)
to keep the repository updated and standardized. The purpose of the

> instruction here is to skeletonize an existing non-skeletonized

Reviewer 2: ```suggestion
instructions here is to describe how to skeletonize an existing non-skeletonized
```

> instructions here is to describe how to skeletonize an existing non-skeletonized

Reviewer: ```suggestion
instructions here are to describe how to skeletonize an existing non-skeletonized
```

Author: Hi @mcdonnnj , grammar text update. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

## General overview ##

The general outline of how to add a skeleton to an existing repository is:

1. [Add the skeleton as a remote](#add-the-skeleton-as-a-remote) to the

> non-skeletonized existing repository

Reviewer 2: ```suggestion
existing non-skeletonized repository
```

> 1. Pull with `--allow-unrelated-histories`

Reviewer: ```suggestion
1. Pull with `git pull --allow-unrelated-histories skeleton-parent`
```

Author: Hi @mcdonnnj , suggested working update for command. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

1. [Fix merge conflicts](#fix-merge-conflicts)
1. [Review non-conflicting changes](#review-non-conflicting-changes) to
prevent merging destructive upstream changes
1. [Update skeleton's `example` references](#update-skeletons-example-references)
1. [Set up pre-commit](#set-up-your-environment-and-pre-commit-hooks)
1. Fix additional problems that may arise
1. [Make a pull request](#make-a-pull-request)

## Add the skeleton as a remote ##

First, decide which of the available skeletons best fits your
existing non-skeletonized repository. To see a list of available
skeletons, use the `gh skeleton list` command or see

> the [list of skeletons](skeleton-list.md).

Reviewer: Are we able to use a link fragment here to link to the appropriate section?

Author: Hi @mcdonnnj , Can you please elaborate regarding the fragment link required ? From my understanding the 'gh skeleton list' command is to be run at the command line. And the skeleton-list link goes to skeleton-list.md document.

Author: Hi @mcdonnnj , update to fragment link. Addressed at 077cfbf

pre-commit install
```

> ## Fix all the inevitable merge conflicts ##

Reviewer 3: Somehow, this comment didn't get recorded when I put in my review.  I think this change should be reverted to the original text and the link change above that references this heading should also be reverted.

Reviewer 2: Fixed in commit ae5f047.

Author: Hi Dave @dav3r, did we resolve the conversation? Also, I'm not sure if I have permissions to merge the PR?

Reviewer 3: Yes, thank you for making the change in https://github.com/cisagov/development-guide/commit/ae5f047ebfc0ea7687ffecd1ee132cecc0d951df.  If you don't have permissions to merge this PR, one of the members of our dev team can take care of it.

## Update skeleton's `example` references ##

Some skeletons need additional configuration. For example, with
`skeleton-python-library` and its module structure inside `src/example`.

This step includes such activities as:

- Update `setup.py` with non-example information
- Arrange into appropriate folders, such as `src` and `test`

> - Update the `codeowners` file to reflect subject matter expertise and

Reviewer: ```suggestion
- Update the `.github/CODEOWNERS` file to reflect subject matter expertise and
```

Author: Hi @mcdonnnj , text wording update. Addressed at 
https://github.com/cisagov/development-guide/pull/97/commits/ffbea4a5f404ec7840245a4eb2f9cb9edbb613b0

## Update skeleton's `example` references ##

Some skeletons need additional configuration. For example, with
`skeleton-python-library` and its module structure inside `src/example`.

This step includes such activities as:

- Update `setup.py` with non-example information
- Arrange into appropriate folders, such as `src` and `test`
- Update the `codeowners` file to reflect subject matter expertise and
codebase familiarity
  - Aim to have at least two codeowners for every repository

## Run pre-commit against existing files ##

The skeleton will bring along with it our standard pre-commit hook

> configurations, including linting and other checks, within `setup-env`.

Reviewer 2: The previous language was correct:
```suggestion
configurations, including linting and other checks, with `setup-env`.
```
